community/planning/griddle/openapi.yaml

```diff
@@ -0,0 +1,2144 @@
+# Copyright 2021 Cargill Incorporated
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+openapi: 3.0.0
+info:
+  version: 0.4.0
+  title: Grid Daemon REST API
+  description: A REST API providing HTTP/JSON interface to Hyperledger Grid.
+  contact:
+    name: Hyperledger Grid community
+    url: https://grid.hyperledger.org/community/
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0
+externalDocs:
+  description: |
+    For more information about how to create Sabre batches, please see the
+    Sabre documentation.
+  url: https://sawtooth.hyperledger.org/docs/sabre/releases/latest/
+paths:
+  # Transactions
+  /statuses:
+    get:
+      tags:
+        - Transaction
+      summary: Fetches the statuses for a set of batches
+      description: |
+        Fetches an array of objects with a status and id for each batch
+        requested. There are four possible statuses with string values
+        `'COMMITTED'`, `'INVALID'`, `'PENDING'`, and `'UNKNOWN'`.
+        The batch(es) you want to check can be specified using the `id` filter
+        parameter. If a `wait` time is specified in the URL, the API will wait
+        to respond until all batches are committed, or the time in seconds has
+        elapsed. If the value of `wait` is not set (i.e., `?wait&id=...`), or
+        it is set to any non-integer value other than `false`, the wait time
+        will be just under the API's specified timeout (usually 300).
+        Note that because this route does not return full resources, the
+        response will not be paginated, and there will be no `head` or
+        `paging` properties.
+      operationId: get_batch_statuses_by_id
+      parameters:
+        - name: id
+          in: query
+          description: A comma-separated list of batch IDs or data change IDs
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/wait"
+        - $ref: "#/components/parameters/service_id"
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/BatchStatusResponse"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "500":
```

Comment:
This endpoint should handle 404s (which happen if given a bad service id). I added a story for that here: https://github.com/hyperledger/grid/issues/1221) which could be added here

Reply:
updated

Reply:
Wait, did this actually get updated? Not seeing it in the file here

Reply:
actually updated now